sphinx_gallery/gen_rst.py

```diff
@@ -231,6 +235,19 @@ def text2string(content):
         return content
 
 
+def extract_thumbnail_number(text):
+    """ Pull out the thumbnail image number specified in the docstring. """
+
+    # check whether the user has specified a specific thumbnail image
+    pattr = re.compile("^\s*#\s*sphinx_gallery_thumbnail_number\s*=\s*([0-9]+)\s*$", flags=re.MULTILINE)
+    match = pattr.search(text)
+    try:
+        thumbnail_number = int(match.groups()[0])
+    except AttributeError:
+        thumbnail_number = 1 # by default, use the first figure created
+
+    return thumbnail_number
```

Comment:
OK, I removed the config stuff entirely for now and moved the thumbnail stuff out of `split_code_and_text_blocks` (it was in there initially so it could remove the comment from the docstring)